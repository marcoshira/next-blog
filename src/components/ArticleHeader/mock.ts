import { PostMetadata } from '../../shared-types/postAttributes';

export default {
  id: '1',
  attributes: {
    title: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    slug: 'teste-1',
    excerpt:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse auctor fermentum nisl, ut pharetra leo lacinia et. Etiam nec urna sed orci bibendum sodales.',
    allowComments: true,
    createdAt: '2022-03-25T18:18:34.944Z',
    updatedAt: '2022-03-27T20:41:54.493Z',
    tags: {
      data: [
        {
          id: '1',
          attributes: {
            displayName: 'UseContext',
            slug: 'use-context',
          },
        },
        {
          id: '3',
          attributes: {
            displayName: 'UseState',
            slug: 'use-state',
          },
        },
      ],
    },
    categories: {
      data: [
        {
          id: '1',
          attributes: {
            displayName: 'JavaScript',
            slug: 'java-script',
          },
        },
        {
          id: '4',
          attributes: {
            displayName: 'React',
            slug: 'react',
          },
        },
      ],
    },
    author: {
      data: {
        id: '4',
        attributes: {
          displayName: 'Júlia Temari',
          slug: 'julia-temari',
        },
      },
    },
    cover: {
      data: {
        id: '14',
        attributes: {
          url: 'https://res.cloudinary.com/dr5ot2zqr/image/upload/v1648156169/1920x1080_6aaecb929e.jpg',
          alternativeText: '1920x1080 1',
        },
      },
    },
    content:
      '<p>We’ll start with some background on the Mac transition and give an overview of Firefox internals that needed to know about the new architecture, before moving on to the concept of Universal Binaries.</p><figure class="table"><table><thead><tr><th>Title 1</th><th>Title 2</th><th>Title 3</th><th>Title 4</th><th>Title 5</th></tr></thead><tbody><tr><td>Content 1</td><td>Content 2</td><td>Content 3</td><td>Content 3</td><td>Content 3</td></tr><tr><td>Content 1</td><td>Content 2</td><td>Content 3</td><td>Content 3</td><td>Content 3</td></tr><tr><td>Content 1</td><td>Content 2</td><td>Content 3</td><td>Content 3</td><td>Content 3</td></tr></tbody></table></figure><blockquote><p><span style="background-color:rgb(255,255,255);color:rgb(0,0,0);">Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec sagittis nisl sagittis lectus ornare aliquet. Fusce faucibus, ex quis facilisis congue, magna elit congue nulla,</span></p></blockquote><p>&nbsp;</p><figure class="image"><img src="https://source.unsplash.com/random/800x800"></figure><p>&nbsp;</p><figure class="image image-style-side"><img src="https://res.cloudinary.com/dr5ot2zqr/image/upload/v1647892277/800x800_a1b3864dfd.jpg" alt="imagem 9"></figure><p><span style="background-color:rgb(255,255,255);color:rgb(0,0,0);">Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec sagittis nisl sagittis lectus ornare aliquet. Fusce faucibus. Imagem ao lado:&nbsp;</span>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p><span style="background-color:rgb(255,255,255);color:rgb(0,0,0);">Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec sagittis nisl sagittis lectus ornare aliquet. Fusce faucibus, ex quis facilisis congue, magna elit congue nulla, </span>Teste de imagem centralizada</p><p>&nbsp;</p><figure class="image"><img src="https://res.cloudinary.com/dr5ot2zqr/image/upload/v1647892277/800x800_a1b3864dfd.jpg" alt="imagem 9"></figure><p><a href="https://google.com.br">Teste de link</a>&nbsp;</p><p><span style="background-color:rgb(255,255,255);color:rgb(0,0,0);"><i>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec sagittis nisl sagittis lectus ornare aliquet. Fusce faucibus, ex quis facilisis congue, magna elit congue nulla,</i></span></p><p><span style="background-color:rgb(255,255,255);color:rgb(0,0,0);"><strong>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec sagittis nisl sagittis lectus ornare aliquet.</strong></span></p><p><code><strong>Teste de código asdasd asad</strong></code></p><pre><code class="language-plaintext">import { PostMetadata } from \'../../shared-types/postAttributes\';\nimport { ArticleHeader } from \'../ArticleHeader\';\nimport { HtmlContent } from \'../HtmlContent\';\nimport * as Styled from \'./styles\';\n\nexport type PostProps = PostMetadata;\n\nexport const Post = ({ attributes }: PostProps) =&gt; {\n  return (\n    &lt;Styled.Wrapper&gt;\n      &lt;ArticleHeader {...attributes} /&gt;\n      &lt;Styled.Content&gt;\n        &lt;HtmlContent html={attributes.content} /&gt;\n      &lt;/Styled.Content&gt;\n    &lt;/Styled.Wrapper&gt;\n  );\n};</code></pre><p><strong>Teste de vídeo</strong></p><figure class="media"><oembed url="https://www.youtube.com/watch?v=heRRKDQepy8"></oembed></figure>',
  },
} as PostMetadata;
