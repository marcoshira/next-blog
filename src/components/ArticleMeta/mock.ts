import { ArticleMetaProps } from '.';

export default {
  createdAt: '2022-03-25T18:28:16.052Z',
  updatedAt: '2022-03-28T18:29:04.677Z',
  tags: {
    data: [
      {
        id: '2',
        attributes: {
          displayName: 'UseRef',
          slug: 'use-ref',
        },
      },
      {
        id: '4',
        attributes: {
          displayName: 'Postgres',
          slug: 'postgres',
        },
      },
    ],
  },
  categories: {
    data: [
      {
        id: '5',
        attributes: {
          displayName: 'Django',
          slug: 'django',
        },
      },
    ],
  },
  author: {
    data: {
      id: '2',
      attributes: {
        displayName: 'Lu√≠sa Santiago',
        slug: 'luisa-santiago',
      },
    },
  },
} as ArticleMetaProps;
