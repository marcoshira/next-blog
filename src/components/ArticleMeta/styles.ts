import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.mediumGray};
    font-style: italic;
    a {
      color: ${theme.colors.crimson};
      text-decoration: none;
      transition: all 300ms ease-in-out;
      &:hover {
        filter: brightness(50%);
      }
    }
  `}
`;
