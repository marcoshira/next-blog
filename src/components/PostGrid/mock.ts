import { PostsProps } from '../../shared-types/postAttributes';

export default {
  posts: {
    data: [
      {
        id: '2',
        attributes: {
          title: 'Teste 2',
          slug: 'teste-2',
          excerpt: 'Teste de post 2.',
          allowComments: false,
          createdAt: '2022-03-25T18:28:16.052Z',
          updatedAt: '2022-03-25T18:29:04.677Z',
          tags: {
            data: [
              {
                id: '2',
                attributes: {
                  displayName: 'UseRef',
                  slug: 'use-ref',
                },
              },
              {
                id: '4',
                attributes: {
                  displayName: 'Postgres',
                  slug: 'postgres',
                },
              },
            ],
          },
          categories: {
            data: [
              {
                id: '5',
                attributes: {
                  displayName: 'Django',
                  slug: 'django',
                },
              },
            ],
          },
          author: {
            data: {
              id: '2',
              attributes: {
                displayName: 'Luísa Santiago',
                slug: 'luisa-santiago',
              },
            },
          },
          cover: {
            data: {
              id: '14',
              attributes: {
                url: 'https://res.cloudinary.com/dr5ot2zqr/image/upload/v1648156169/1920x1080_6aaecb929e.jpg',
                alternativeText: '1920x1080 1',
              },
            },
          },
          content:
            '<p>Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.</p><p>To wrap up January, we are proud to bring you the release of Firefox 85. In this version we are bringing you support for the :focus-visible pseudo-class in CSS and associated devtools, and the complete removal of Flash support from Firefox.</p>',
        },
      },
      {
        id: '1',
        attributes: {
          title: 'Teste 1',
          slug: 'teste-1',
          excerpt: 'Teste de excerpt.',
          allowComments: true,
          createdAt: '2022-03-25T18:18:34.944Z',
          updatedAt: '2022-03-27T20:41:54.493Z',
          tags: {
            data: [
              {
                id: '1',
                attributes: {
                  displayName: 'UseContext',
                  slug: 'use-context',
                },
              },
              {
                id: '3',
                attributes: {
                  displayName: 'UseState',
                  slug: 'use-state',
                },
              },
            ],
          },
          categories: {
            data: [
              {
                id: '1',
                attributes: {
                  displayName: 'JavaScript',
                  slug: 'java-script',
                },
              },
              {
                id: '4',
                attributes: {
                  displayName: 'React',
                  slug: 'react',
                },
              },
            ],
          },
          author: {
            data: {
              id: '4',
              attributes: {
                displayName: 'Júlia Temari',
                slug: 'julia-temari',
              },
            },
          },
          cover: {
            data: {
              id: '14',
              attributes: {
                url: 'https://res.cloudinary.com/dr5ot2zqr/image/upload/v1648156169/1920x1080_6aaecb929e.jpg',
                alternativeText: '1920x1080 1',
              },
            },
          },
          content:
            '<p>We’ll start with some background on the Mac transition and give an overview of Firefox internals that needed to know about the new architecture, before moving on to the concept of Universal Binaries.</p><figure class="table"><table><thead><tr><th>Title 1</th><th>Title 2</th><th>Title 3</th><th>Title 4</th><th>Title 5</th></tr></thead><tbody><tr><td>Content 1</td><td>Content 2</td><td>Content 3</td><td>Content 3</td><td>Content 3</td></tr><tr><td>Content 1</td><td>Content 2</td><td>Content 3</td><td>Content 3</td><td>Content 3</td></tr><tr><td>Content 1</td><td>Content 2</td><td>Content 3</td><td>Content 3</td><td>Content 3</td></tr></tbody></table></figure><blockquote><p>Teste de citação</p></blockquote><p>&nbsp;</p><p>Teste de foto ao lado</p><figure class="image"><img src="https://res.cloudinary.com/dr5ot2zqr/image/upload/v1647892307/800x800_74f76986f2.jpg" alt="imagem 11"></figure><p>&nbsp;</p><p>Teste de imagem centralizada</p><p>&nbsp;</p><p>&nbsp;</p><figure class="image"><img src="https://res.cloudinary.com/dr5ot2zqr/image/upload/v1647892277/800x800_a1b3864dfd.jpg" alt="imagem 9"></figure><p><a href="https://google.com.br">Teste de link</a></p><p>&nbsp;</p><p><i>Itálico</i></p><p>&nbsp;</p><p><strong>Negrito</strong></p><p>&nbsp;</p><p><code><strong>Teste de código</strong></code></p><p>&nbsp;</p><pre><code class="language-plaintext">Teste de bloco de código</code></pre><p>&nbsp;</p><p><strong>Teste de vídeo</strong></p><p>&nbsp;</p><figure class="media"><oembed url="https://www.youtube.com/watch?v=qF1DTK4U1AM"></oembed></figure>',
        },
      },
    ],
  },
} as PostsProps;
