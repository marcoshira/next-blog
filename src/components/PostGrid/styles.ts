import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css``}
`;

export const Post = styled.div`
  ${({ theme }) => css``}
`;

export const NotFound = styled.div`
  ${({ theme }) => css``}
`;
