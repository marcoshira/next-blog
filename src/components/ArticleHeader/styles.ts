import styled, { css } from 'styled-components';
import { Wrapper as ArticleMeta } from '../ArticleMeta/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    > h1 {
      margin-bottom: ${theme.spacings.medium};
      max-width: 80rem;
    }

    > p {
      margin-top: 0;
      max-width: 80rem;
    }

    > img {
      width: 100%;
      max-width: 80rem;
    }

    & ${ArticleMeta} {
      max-width: 80rem;
    }
  `}
`;
