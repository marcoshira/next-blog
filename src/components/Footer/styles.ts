import styled, { css } from 'styled-components';
import { Container as HtmlContent } from '../HtmlContent/styles';

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.2rem solid ${theme.colors.mediumGray};

    a {
      color: inherit;
      text-decoration: none;
    }

    & ${HtmlContent} {
      font-size: ${theme.font.sizes.small};

      > p > a {
        color: ${theme.colors.black};
      }
    }
  `}
`;
