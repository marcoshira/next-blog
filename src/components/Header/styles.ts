import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';

type DisplayText = {
  showText?: boolean;
};

const displayText = () => css`
  display: flex;
`;

export const Wrapper = styled.header`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    align-content: center;
    justify-content: center;
    width: 100%;

    div {
      margin: ${theme.spacings.xlarge} ${theme.spacings.medium};
    }
  `}
`;

export const Container = styled.div<DisplayText>`
  ${({ theme, showText }) => css`
    display: none;
    ${showText && displayText()}
    flex-direction: column;
    max-width: 70rem;

    & ${Heading} {
      margin-top: 0;
      margin-bottom: ${theme.spacings.xsmall};
    }
  `}
`;
