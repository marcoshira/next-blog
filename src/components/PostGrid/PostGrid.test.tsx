import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { PostGrid, PostGridProps } from '.';
import mock from './mock';

const mockPosts = {
  posts: {
    data: [],
  },
};

const props: PostGridProps = mock.posts;

describe('<PostGrid />', () => {
  it('should render', () => {
    const { container } = renderTheme(<PostGrid {...props} />);

    expect(container).toMatchSnapshot();
  });
  it('should render without posts', () => {
    renderTheme(<PostGrid {...mockPosts.posts} />);

    expect(screen.getByText('Nenhum post encontrado')).toBeInTheDocument();
  });
});
