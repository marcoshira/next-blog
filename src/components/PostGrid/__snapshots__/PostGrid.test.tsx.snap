// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostGrid /> should render 1`] = `
.c0 {
  width: 100%;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(28rem,1fr));
  gap: 3.2rem;
  padding: 3.2rem;
}

.c5 {
  color: inherit;
  font-size: 2.4rem;
  text-transform: none;
}

.c2 .c4 {
  margin: 0;
  margin-top: 1.6rem;
  margin-bottom: 0.8rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c2 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c2:hover .c4 {
  color: crimson;
}

.c2 img {
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c2:hover img {
  opacity: 0.6;
}

.c2:hover p {
  color: #AAAAAA;
}

.c3 {
  max-width: 100%;
}

.c6 {
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

@media (max-width:440px) {
  .c1 {
    grid-template-columns: 1fr;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <a
          href="/post/teste-2"
        >
          <img
            alt="Teste 2"
            class="c3"
            src="https://images.unsplash.com/photo-1649940277866-6de246635a96?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=800&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTY1MDg0OTY1OQ&ixlib=rb-1.2.1&q=80&w=800"
          />
          <h3
            class="c4 c5"
          >
            Teste 2
          </h3>
          <p
            class="c6"
          >
            Teste de post 2.
          </p>
        </a>
      </div>
      <div
        class="c2"
      >
        <a
          href="/post/teste-1"
        >
          <img
            alt="Teste 1"
            class="c3"
            src="https://res.cloudinary.com/dr5ot2zqr/image/upload/v1648156169/1920x1080_6aaecb929e.jpg"
          />
          <h3
            class="c4 c5"
          >
            Teste 1
          </h3>
          <p
            class="c6"
          >
            Teste de excerpt.
          </p>
        </a>
      </div>
    </div>
  </div>
</div>
`;
